pkg/virt-operator/resource/generate/components/crds.go

```diff
@@ -766,6 +766,17 @@ func NewPrometheusRuleSpec(ns string) *promv1.PrometheusRuleSpec {
 							"summary": "More than 80% of the rest calls failed in virt-handler for the last 5 minutes",
 						},
 					},
+					{
+						Alert: "OrphanedVirtualMachineImages",
+						Expr:  intstr.FromString("(count by(node)(node_namespace_pod:kube_pod_info:{pod=~'virt-launcher-.*'} ) * on (node) group_left(pod) (1*(kube_pod_container_status_ready{pod=~'virt-handler-.*'} + on (pod) group_left(node) (0 * node_namespace_pod:kube_pod_info:{pod=~'virt-handler-.*'} ))) or on (node) (0 * node_namespace_pod:kube_pod_info:{pod=~'virt-launcher-.*'} ) ) == 0"),
```

Comment:
@ashleyschuett can you perhaps add a comment here to explain the formula you're using here? I kind of follow it, but I'm not 100% sure I understand the intent. I think the parts where you have `(0 * something)`  confuses me the most.

Reply:
@davidvossel  Hi, The (0*something) is to be able to do a join but not affect the value. In Prometheus it all ends up to be arithmetic calculations on the value. If you only want the value of one of them , you need to make sure the others are either 0 or 1 based on the mathematical operation you are doing. 

1. First you are counting the number of virt-launcher pods in the node. 
2. Then join it (multiply by) based on the node with the number of ready virt-handler pods (use its value) joined with the number of virt-handler pods on the node(not use its value) "or" join on the number of virt-launcher pods (not use its value)  when there are no virt-handler pods on the node so that the join will not drop the record.

Reply:
@davidvossel would you like this commented in the code as well?

Reply:
I'm still struggling a bit. Maybe this would help me... What prevents this expression from equaling `0` when there are no launchers and no virt-handlers on a node?

Reply:
@ashleyschuett I would consider adding the calculation as a recording rule and then use it in the alert rule. The recording rule allows you to precompute frequently needed or computationally expensive expressions and save their result as a new set of time series.

Reply:
@davidvossel If there are no launchers and no virt-handlers on a node the record should drop completely. @ashleyschuett did you test this scenario?

Reply:
@sradco yes I did. As you statued the record is completely dropped. Should I add the whole query as a recording rule, or break it up into two separate recording rules for the query to be more optimal? 

```
(count by(node)(node_namespace_pod:kube_pod_info:{pod=~'virt-launcher-.*'} ) * on (node) group_left(pod) (1*(kube_pod_container_status_ready{pod=~'virt-handler-.*'} + on (pod) group_left(node) (0 * node_namespace_pod:kube_pod_info:{pod=~'virt-handler-.*'} ))) or on (node)
```

and 
```
(0 * node_namespace_pod:kube_pod_info:{pod=~'virt-launcher-.*'} )
```